front/app/modules/commercial/analytics/admin/hooks/useVisitorsData/parse/index.ts

```diff
@@ -0,0 +1,117 @@
+// parse dates
+import { parseMonths } from './parseMonths';
+import { parseWeeks } from './parseWeeks';
+import { parseDays } from './parseDays';
+
+// utils
+import { keys } from 'utils/helperUtils';
+import { roundDateToMidnight } from './utils';
+
+// typings
+import { Moment } from 'moment';
+import { IResolution } from 'components/admin/ResolutionControl';
+import { Response, Stats, TimeSeries, TimeSeriesResponse } from '../typings';
+import { Translations } from '../utils';
+
+export const parseStats = ([
+  totalsWholePeriodRows,
+  totalsLastPeriodRows,
+]: Response['data']): Stats => {
+  const wholePeriod = totalsWholePeriodRows[0];
+  const lastPeriod = totalsLastPeriodRows[0];
+
+  return {
+    visitors: {
+      value: wholePeriod?.count_visitor_id.toLocaleString() ?? '0',
+      lastPeriod: lastPeriod?.count_visitor_id.toLocaleString() ?? '0',
+    },
+    visits: {
+      value: wholePeriod?.count.toLocaleString() ?? '0',
+      lastPeriod: lastPeriod?.count.toLocaleString() ?? '0',
+    },
+    visitDuration: {
+      value: wholePeriod?.avg_duration ?? '-',
+      lastPeriod: lastPeriod?.avg_duration ?? '-',
+    },
+    pageViews: {
+      value: wholePeriod?.avg_pages_visited ?? '-',
+      lastPeriod: lastPeriod?.avg_pages_visited ?? '-',
+    },
+  };
+};
+
+export const parseTimeSeries = (
+  responseTimeSeries: TimeSeriesResponse,
+  startAtMoment: Moment | null | undefined,
+  endAtMoment: Moment | null | undefined,
+  resolution: IResolution
+): TimeSeries | null => {
+  if (responseTimeSeries.length === 0) return null;
+
+  const startAtMomentRounded = startAtMoment
+    ? roundDateToMidnight(startAtMoment)
+    : startAtMoment;
```

Comment:
The date picker we have returns a `Moment` object where the time is 12 in the afternoon. Here I am just making sure that all `Moment`s we use are always rounded to 12 at night, which is the default if you parse a date like `moment('2022-09-01')`